gtsam/discrete/TableFactor.cpp

```diff
@@ -64,27 +64,69 @@ TableFactor::TableFactor(const DiscreteKeys& dkeys,
 /**
  * @brief Compute the correct ordering of the leaves in the decision tree.
  *
- * This is done by first taking all the values which have modulo 0 value with
- * the cardinality of the innermost key `n`, and we go up to modulo n.
- *
  * @param dt The DecisionTree
- * @return std::vector<double>
+ * @return Eigen::SparseVector<double>
```

Comment:
This function says it computes an ordering but then returns a sparse vector. Maybe add more documentation and possibly even change the name if it no longer reflects the code.